### PR TITLE
Fix chapter descriptions, again

### PR DESCRIPTION
**Pull request by felipecalvo**
## :dart: Goal

Fix a bug introduced in #1574 and some other _very_ minor visual inconveniences.

There are a few refactors but no big visual changes. Everything looks the same except for a few small things (see `Details`).

## :memo: Details

I came to the conclusion that the bug I introduced - as well as the one I fixed in #1574 in the first place - wouldn't have occurred if I had found the layout easier to understand. We rely on `pull-left` too much and there are some `div`s of height 0 due to the way the view is structured. So in order to make it clearer, I:

* Modified headers (that is, title + description) for chapter, guide and exercise so that they share a more similar structure, even though the code is not extracted
* Moved around a couple `.html.erb` files
* Favored using flex whenever possible, replacing all those `pull-left`
* Deleted a few unused CSS classes

Other small oversights I fixed:
* Chapter descriptions were not justified, unlike guide and exercise description texts.
* The teacher info/edit icons were too close to the title.  (Before/after)
Screenshot from 2021-02-27 01-57-54
Screenshot from 2021-02-27 02-02-14
* Language icon didn't hide on mobile guide view (it does on exercise view), which looked a bit ugly sometimes, plus the other icons were wrapping wrongly. (Before/after)
Screenshot from 2021-02-27 02-10-45

## :camera_flash: Screenshots

And now a screenshot for every case I tested

### Chapter (long description)

image
____

### Chapter (short description)
image
____

### Chapter (mobile view)

____

### Chapter (monolesson; lesson has medal and teacher info)
image

____

### Chapter (monolesson; lesson has medal and teacher info; mobile view)

____

### Guide (short description)
image

____

### Guide (long description, medal, teacher info)
image

____

### Guide (mobile view)

____

### Exercise
image

____

## :back: Backwards compatibility

All of it

### CODE REVIEW

File: app/helpers/content_view_helper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  private
+
+  def content_type_number(content)
+    "#{content_type(content)}_number"
+  end
+
+  def content_type(content)
+    content.is_a?(Exercise) ? "exercise" : content.class.name.downcase
+  end
```
Comment (felipecalvo) [inline app/helpers/content_view_helper.rb:18]: Me trying to get line 17 approved

image

Comment (flbulgarelli) [reply app/helpers/content_view_helper.rb:18]: It's not that bad! Just check if model_name already does that

**Review by flbulgarelli — APPROVED** [review]

File: app/helpers/content_view_helper.rb (hunk 1 above)
Comment (felipecalvo) [reply app/helpers/content_view_helper.rb]: That's great, it does work!

Comment (luchotc) [reply app/helpers/content_view_helper.rb]: image

**Review by luchotc — APPROVED** [review]
@felipecalvo This is SO great! :clap: :clap: :clap:

**Pull request merged**